packages/ui/src/accounts/hooks/useIsVoteStakeLocked.ts

```diff
@@ -6,5 +6,5 @@ export const useIsVoteStakeLocked = (
 ) => {
   // Lock stake if the vote was cast: in current election or to winning candidate
   // Enable stake recovery if election is finished
-  return isLatestElection && !!candidate && (!isElectionFinished || candidate.isCouncilMember)
+  return isLatestElection && (!isElectionFinished || !!candidate?.isCouncilMember)
```

Comment:
Before this change, all votes in _Voting period_ and sealed votes in _Reveal period_ have undefined `voteFor` field so useVoteStakeLocked returned false even though stake shouldn't be recoverable

Reply:
Nice catch! Plus the logic is a lot more readable this way.

But I think for it to work fully in [@/accounts/components/AccountItem/components/LocksDetails.tsx:31](https://github.com/Joystream/pioneer/blob/4231911c6abf17d34a54a969cf5b9a23b9da0337/packages/ui/src/accounts/components/AccountItem/components/LocksDetails.tsx#L31) the line should be changed to:
```ts
const isVoteStakeLocked = useIsVoteStakeLocked(candidate, {
  isLatestElection: !!candidate,
  isElectionFinished: election?.isFinished,
})
```
Also I just realized the comment is wrong in `useIsVoteStakeLocked`, AFAICS it should be:
```ts
// Always allow stake recovery for votes not cast in the latest election
// also allow when both the election is over and the voted for candidate lost
```

Reply:
Thanks for the comment in `useIsVoteStakeLocked`, it's much better now